### DISCUSSION TITLE
How to get SQLpage version without running SQLpage?

### DISCUSSION THREAD

**Discussion by lyderic** [sqlpage/SQLPage#562] [Q&A]
I am writing a script to update SQLpage when a new version is released. I would like to check the installed version so that I run the rest of the script only if versions differ.

The sqlpage.bin binary doesn't seem to have any command parameters:

```bash
$ sqlpage --help
$ sqlpage --version
```

Whatever I try, it always launches the server with some defaults (creates sqlpage directory, runs on port 8080 etc.)

I know that when the server is running, I can get the version information, for example by creating the following version.sql file:

```sql
SELECT 'json' AS component, sqlpage.version() AS contents;
```

However, I can't use this method in a script.

So far, I am doing this:

```bash
#!/bin/bash
[ -z ${1} ] && {
  echo "version argument missing!"
  exit 1
}
version=${1}
[ -x /usr/local/bin/sqlpage ] && {
  installed_version=$(strings /usr/local/bin/sqlpage \
    | grep 'started successfully' \
    | cut -d'v' -f3 \
    | awk '{print $1}') 
  [ ${installed_version} == ${version} ] && {
    echo "SQLpage version ${version} already installed"
    exit 0
  }
}
# ... rest of the script
```

As you can see, I have to resort to digging and parsing an ASCII string from the executable! Is there a better, more reliable way to get SQLpage version from a script?

**Comment by lovasoa** [comment]
That would be a useful feature! Would you like to implement it in SQLPage?

**Reply by lyderic** [reply]
Though I don't know rust, I can give it a try. I'll submit a PR when I have some time to look at it.
L.

**Comment by lyderic** [comment]
Well I must have done something wrong, because implementing command line arguments was surprisingly easier than I thought ;-)

I will submit a PR soon.
